tonic/models/vic/grid_params.py

```diff
@@ -1648,3 +2150,369 @@ def lake(lake_file, soil_dict, max_numnod=10,
+        nbands_expected = len(bands_present)
+
+        # Make sure state file has correct number of snow bands
+        if nbands_state[cell] != nbands_expected:
+            raise ValueError('nbands in state file != nbands expected based on snow bands file')
+
+        # Continue parsing state file
+        if cell == 0:
+            for i in range(0, soil_nodes):
+                dz_node[i] = temp[i + 3]
+            for i in range(0, soil_nodes):
+                node_depth[i] = temp[i + 3 + soil_nodes]
+        numrows = nveg_state[cell] * nbands_state[cell] + row + 1
+        row += 1
+
+        while row < numrows:
+            lines[row] = lines[row].strip()
+            line = lines[row].strip('\n').split(' ')
+            temp = np.array(line).astype(float)
+            iveg = int(temp[0])
+            iband = int(temp[1])
+            veg_class = veg_classes_present[iveg]
+            band = bands_present[iband]
+
+            tmp = 2
+            Soil_moisture[cell, veg_class, band, :] = temp[tmp:tmp + nlayers]
+            tmp += nlayers
+            for lidx in range(nlayers):
+                Soil_ice[cell, veg_class, band, lidx, :] = temp[tmp:tmp + nfrost]
+                tmp += nfrost
+            if iveg < nveg_state[cell]-1 or bare[cell] == 0:
+                Canopy_water[cell, veg_class, band] = temp[tmp]
+                tmp += 1
+            else:
+                Canopy_water[cell, veg_class, band] = 0
+
+            if carbon:
+                if iveg < nveg_state[cell]-1 or bare[cell] == 0:
+                    AnnualNPP[cell, veg_class, band] = temp[tmp]
+                    AnnualNPPPrev[cell, veg_class, band] = temp[tmp + 1]
+                    CLitter[cell, veg_class, band] = temp[tmp + 2]
+                    CInter[cell, veg_class, band] = temp[tmp + 3]
+                    CSlow[cell, veg_class, band] = temp[tmp + 4]
+                    tmp += 5
+                else:
+                    AnnualNPP[cell, veg_class, band] = 0
+                    AnnualNPPPrev[cell, veg_class, band] = 0
+                    CLitter[cell, veg_class, band] = 0
+                    CInter[cell, veg_class, band] = 0
+                    CSlow[cell, veg_class, band] = 0
+
+            Snow_age[cell, veg_class, band] = int(temp[tmp])
+            Snow_melt_state[cell, veg_class, band] = int(temp[tmp + 1])
+            Snow_coverage[cell, veg_class, band] = temp[tmp + 2]
+            Snow_water_equivalent[cell, veg_class, band] = temp[tmp + 3]
+            Snow_surf_temp[cell, veg_class, band] = temp[tmp + 4]
+            Snow_surf_water[cell, veg_class, band] = temp[tmp + 5]
+            Snow_pack_temp[cell, veg_class, band] = temp[tmp + 6]
+            Snow_pack_water[cell, veg_class, band] = temp[tmp + 7]
+            Snow_density[cell, veg_class, band] = temp[tmp + 8]
+            Snow_cold_content[cell, veg_class, band] = temp[tmp + 9]
+            Snow_canopy[cell, veg_class, band] = temp[tmp + 10]
+            tmp += 11
```

Comment:
A lot of this would tighten up nicely if you used an `OrderedDict`:

```python
for offset, (key, array) in enumerate(state_dict.items()):
    state_dict[key] = int(temp[tmp + offset])
```

Obviously you'll have to think about the type casting at some point too but that can probably be done when you create the arrays.